hack/update-generated-crd.sh

```diff
@@ -1,11 +1,25 @@
 #!/bin/bash
 set -euo pipefail
 
-VENDORED_CRD='vendor/github.com/openshift/api/operator/v1/0000_70_dns-operator_00.crd.yaml'
-LOCAL_CRD='manifests/0000_70_dns-operator_00.crd.yaml'
-
-if [[ -z "${SKIP_COPY+1}" ]]; then
-  if ! cmp -s "$LOCAL_CRD" "$VENDORED_CRD"; then
-    cp -f "$VENDORED_CRD" "$LOCAL_CRD"
+function install_crd {
+  if [[ -z "${SKIP_COPY+1}" ]]; then
+    local SRC="$1"
+    local DST="$2"
+    if ! cmp -s "$DST" "$SRC"; then
+      cp -f "$SRC" "$DST"
+    fi
   fi
-fi
+}
+
+# Can't rely on associative arrays for old Bash versions (e.g. OSX)
+install_crd \
+  "vendor/github.com/openshift/api/operator/v1/0000_70_dns-operator_00.crd.yaml" \
+  "manifests/0000_70_dns-operator_00.crd.yaml"
+
+install_crd \
+  "vendor/github.com/openshift/api/network/v1alpha1/0000_70_dnsnameresolver_00-techpreview.crd.yaml" \
+  "pkg/manifests/assets/dnsnameresolver/0000_70_dnsnameresolver_00-techpreview.crd.yaml"
+
+install_crd \
+  "vendor/github.com/openshift/api/network/v1alpha1/0000_70_dnsnameresolver_00-customnoupgrade.crd.yaml" \
+  "pkg/manifests/assets/dnsnameresolver/0000_70_dnsnameresolver_00-customnoupgrade.crd.yaml"
```

Comment:
Can you give me some explanation as to why you need to make these changes?  Also why there are two different crd files for techpreview and customnoupgrade?

Reply:
I think you should choose just one of these to simplify things for docs and users.

Reply:
If the CRD corresponding to `CustomNoUpgrade` is not installed then it'll be breaking `CustomNoUpgrade` use cases. This was requested by Joel during the review of the API PR: https://github.com/openshift/api/pull/1524#discussion_r1262678270